src/Chainweb/Pact/Backend/Types.hs

```diff
@@ -83,6 +110,59 @@ instance A.FromJSON PactDbConfig
 
 makeLenses ''PactDbConfig
 
+data SQLiteEnv = SQLiteEnv
+  { _sConn :: Database
+  , _sConfig :: !SQLiteConfig
+  }
+
+makeLenses ''SQLiteEnv
+
+newtype ReorgVersion = ReorgVersion
+  { _getReorgVersion :: Int64
+  }
+  deriving newtype Num
+  deriving stock Show
+
+data BlockVersion = BlockVersion
+  { _bvBlock :: !BlockHeight
+  , _bvVersion :: !ReorgVersion
+  }
+  deriving Show
+
+makeLenses ''BlockVersion
+
+data BlockState = BlockState
+  { _bsTxId :: !P.TxId
+  , _bsMode :: Maybe P.ExecutionMode
+  , _bsBlockVersion :: !BlockVersion
+  }
+
+makeLenses ''BlockState
+
+data CWDbEnv p = CWDbEnv
+  { _cwDb :: p
+  , _cwBlockState :: !BlockState
+  }
+
+makeLenses ''CWDbEnv
+
+type VersionHandler p a = ReaderT p (StateT BlockState IO) a
+
+-- data PactDbStateNew = PactDbStateNew
+--   { _pdsPactDb :: P.PactDb (CWDbEnv SQLiteEnv)
+--   , _pdsdbEnv :: CWDbEnv SQLiteEnv
+--   }
+
+-- makeLenses ''PactDbStateNew
+
+data CheckpointerNew p = CheckpointerNew
+    { restoreNew :: BlockHeight -> BlockHash -> IO (Either String (p, P.TxId, Maybe P.ExecutionMode))
```

Comment:
We don't need TxId as an output anymore, and ExecutionMode is an input to restoreXXX